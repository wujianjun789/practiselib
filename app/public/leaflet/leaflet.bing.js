/**
 * Created by a on 2018/1/9.
 */
