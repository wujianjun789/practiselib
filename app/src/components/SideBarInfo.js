/**
 * Created by a on 2017/7/5.
 */
import React, {Component} from 'react'
import '../../public/styles/sideBarInfo.less';

import Pie from './SensorParamsPie'
import {updateMap, updateMapDevice, destory} from '../util/map'
export default class SideBarInfo extends Component{
    constructor(props){
        super(props)

        this.state = {
            collapse:false,
        }

        this.latlng = {};

        this.renderMap = this.renderMap.bind(this);
        this.collpseHandler = this.collpseHandler.bind(this);
    }

    componentWillMount(){

    }

    componentWillUnmount(){
        destory();
    }

    collpseHandler(){
        this.setState({collapse:!this.state.collapse});

        this.props.collpseHandler && this.props.collpseHandler();
    }

    renderMap(ref) {
        const {mapDevice} = this.props;
        if (ref) {
            updateMap({latlng:this.latlng});
            let key = 'lamp';
            switch(mapDevice.position["device_type"]){
                case "DEVICE":
                    key = 'lamp';
                    break;
                case "CONTROLLER":
                    key = 'controller';
                    break;
                case "ISTREETLIGHT":
                    key = 'intelligent';
                    break;
                case "CHARGER":
                    key = 'charger';
                    break;
            }

            updateMapDevice([mapDevice.position], {[key]:[mapDevice.data]})
        }
    }

    render(){
        const {collapse} = this.state;
        let {deviceInfo={total:0, normal:0}} = this.props;
        const {total, normal} = deviceInfo;
        let width=145,height=145;
        return <div className={"container-fluid sidebar-info "+(collapse ? "sidebar-collapse":"")}>
                <div className="row collapse-container" onClick={()=>this.collpseHandler()}>
                    <span className={collapse ? "icon_horizontal":"icon_verital"}></span>
                </div>
                <div className="panel panel-default device-statics-info">
                    <div className="panel-heading">
                        <span className="icon_statistics"></span>设备统计信息
                    </div>
                    <div className="panel-body view">
                        <div className="circle1">
                            <Pie data={{type:"NOISE",val:total}} width={width} height={height} color="#E6BC00" className="noise" range={[0, total]}></Pie>
                        </div>
                        <div className="circle2">
                            <Pie data={{type:"TEMPS", val:normal, unit:'%'}} width={width} height={height} color="#E6BC00" className="temps" range={[0, total]}></Pie>
                        </div>
                    </div>
                </div>
                <div className="panel panel-default map-position">
                    <div className="panel-heading">
                        <span className="icon_map_position"></span>地图位置
                    </div>
                    <div ref={this.renderMap} id="map" className="panel-body map"></div>
                </div>
        </div>
    }
}