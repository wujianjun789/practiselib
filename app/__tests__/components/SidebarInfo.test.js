/**
 * Created by a on 2017/7/6.
 */
import React from 'react';
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme';

import '../../public/js/jquery-3.1.1.min'
import '../../public/leaflet/leaflet';
window.d3 = require('../../public/js/d3.min')

import SideBarInfo from '../../src/components/SideBarInfo'

test('SidebarInfo renders', ()=>{

    const component = renderer.create(
        <SideBarInfo />
    )

    let sideBarInfo = component.toJSON();
    expect(sideBarInfo).toMatchSnapshot();
})

test('SidebarInfo div click', ()=>{
    const component = shallow(<SideBarInfo />)

    component.find('.collapse-container').simulate('click');
    expect(component.is('.sidebar-collapse')).toEqual(true);
})

