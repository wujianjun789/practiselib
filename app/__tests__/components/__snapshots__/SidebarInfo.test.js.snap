exports[`test SidebarInfo renders 1`] = `
<div
  className="container-fluid sidebar-info ">
  <div
    className="row collapse-container"
    onClick={[Function]}>
    <span
      className="icon_verital" />
  </div>
  <div
    className="panel panel-default device-statics-info">
    <div
      className="panel-heading">
      <span
        className="icon_statistics" />
      设备统计信息
    </div>
    <div
      className="panel-body view">
      <div
        className="circle1">
        <svg
          className="noisePieChart">
          <defs>
            <pattern
              height="400"
              id="noise-img"
              patternUnits="userSpaceOnUse"
              width="400">
              <image
                height="400"
                transform="rotate(97 200 200)"
                width="400"
                x="0"
                xlinkHref="/svg/noise.svg"
                y="0" />
            </pattern>
            <pattern
              height="400"
              id="temps-img"
              patternUnits="userSpaceOnUse"
              width="400">
              <image
                height="400"
                transform="rotate(97 200 200)"
                width="400"
                x="0"
                xlinkHref="/svg/temps.svg"
                y="0" />
            </pattern>
          </defs>
          <g
            className="circle">
            <circle
              className="base-circle" />
            <circle
              className="progress-circle" />
          </g>
          <g
            className="text">
            <text
              className="type"
              x="200"
              y="140" />
            <text
              className="val"
              x="200"
              y="240" />
          </g>
        </svg>
      </div>
      <div
        className="circle2">
        <svg
          className="tempsPieChart">
          <defs>
            <pattern
              height="400"
              id="noise-img"
              patternUnits="userSpaceOnUse"
              width="400">
              <image
                height="400"
                transform="rotate(97 200 200)"
                width="400"
                x="0"
                xlinkHref="/svg/noise.svg"
                y="0" />
            </pattern>
            <pattern
              height="400"
              id="temps-img"
              patternUnits="userSpaceOnUse"
              width="400">
              <image
                height="400"
                transform="rotate(97 200 200)"
                width="400"
                x="0"
                xlinkHref="/svg/temps.svg"
                y="0" />
            </pattern>
          </defs>
          <g
            className="circle">
            <circle
              className="base-circle" />
            <circle
              className="progress-circle" />
          </g>
          <g
            className="text">
            <text
              className="type"
              x="200"
              y="140" />
            <text
              className="val"
              x="200"
              y="240" />
          </g>
        </svg>
      </div>
    </div>
  </div>
  <div
    className="panel panel-default map-position">
    <div
      className="panel-heading">
      <span
        className="icon_map_position" />
      地图位置
    </div>
    <div
      className="panel-body map"
      id="map" />
  </div>
</div>
`;
